rest/route/project.go

```diff
@@ -1048,3 +1048,59 @@ func (h *projectParametersGetHandler) Run(ctx context.Context) gimlet.Responder
 
 	return gimlet.NewJSONResponse(res)
 }
+
+////////////////////////////////////////////////////////////////////////
+//
+// PUT /rest/v2/projects/value/{toReplace}/{replacement}
+
+type rotation struct {
+	toReplace   string
+	replacement string
+}
+
+type projectVarsPatchHandler struct {
+	replaceVars *rotation
+	user        *user.DBUser
+	args        []byte
+
+	sc       data.Connector
+	settings *evergreen.Settings
+}
+
+func makeProjectVarsPatch(sc data.Connector, settings *evergreen.Settings) gimlet.RouteHandler {
+	return &projectVarsPatchHandler{
+		sc:       sc,
+		settings: settings,
+	}
+}
+
+func (h *projectVarsPatchHandler) Factory() gimlet.RouteHandler {
+	return &projectVarsPatchHandler{
+		sc:       h.sc,
+		settings: h.settings,
+	}
+}
+
+// Parse fetches the project's identifier from the http request.
+func (h *projectVarsPatchHandler) Parse(ctx context.Context, r *http.Request) error {
+	h.user = MustHaveUser(ctx)
+	body := util.NewRequestReader(r)
+	b, err := ioutil.ReadAll(body)
+	if err != nil {
+		return errors.Wrap(err, "Argument read error")
+	}
+	h.args = b
+	replacements := &rotation{}
+	if err = json.Unmarshal(b, replacements); err != nil {
+		return errors.Wrap(err, "API error while unmarshalling JSON")
+	}
+	h.replaceVars = replacements
+	defer body.Close()
+	return nil
+}
+
+// Run updates a project by name.
+func (h *projectVarsPatchHandler) Run(ctx context.Context) gimlet.Responder {
+	res, _ := h.sc.UpdateProjectVarsByValue(h.replaceVars.toReplace, h.replaceVars.replacement)
```

Comment:
We should check this error (and as mentioned earlier, a list of modified projects may be more helpful than the MongoDB change info)